Nickvision.Aura/Network/WebHelpers.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Nickvision.Aura.Network;
+
+/// <summary>
+/// A utility class for working with the web
+/// </summary>
+public static class WebHelpers
+{
+    /// <summary>
+    /// A static HttpClient
+    /// </summary>
+    /// <remarks>Microsoft recommends not creating and using HttpClient objects over and over again</remarks>
+    public static HttpClient Client { get; }
+
+    /// <summary>
+    /// Constructs WebHelpers statically
+    /// </summary>
+    static WebHelpers()
+    {
+        Client = new HttpClient();
+    }
+
+    /// <summary>
+    /// Gets whether or not a Uri leads to a valid website
+    /// </summary>
+    /// <param name="uri">Uri</param>
+    /// <returns>True if connection successful, else false</returns>
+    public static async Task<bool> GetIsValidWebsiteAsync(this Uri uri) => (await Client.GetAsync(uri)).StatusCode != HttpStatusCode.NotFound;
```

Comment:
Will it be `NotFound` if there's no connection?

Reply:
I think it will actually throw...either way made it return false which will just say no update in UI terms